lightning/src/ln/channelmanager.rs

```diff
@@ -2506,7 +2526,9 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					},
 					None => { insert_outbound_payment!(); },
 				}
-			} else { unreachable!(); }
+			} else {
+				return Err(APIError::ChannelUnavailable{err: "No channel available with first hop!".to_owned()});
```

Comment:
Hmm I'm not sure if you mean that we'd like to change the error message here to indicate to the user the lack between  the `short_to_chan_info` and the `by_id` maps, or if you mean that we should just add a code comment here to indicate it. I don't think we should change the message to surface the reason for the problem to the user, as neither the `short_to_chan_info` nor the `by_id` map are public to the user, and it would therefore make no sense to refer to them in the message.

I'll add a code comment here to note why this was triggered :)